feat(eslint-plugin): [no-base-to-string] add support for catching toLocaleString (#10138)
typescript-eslint/typescript-eslint: packages/eslint-plugin/docs/rules/no-base-to-string.mdx

>
> See **https://typescript-eslint.io/rules/no-base-to-string** for documentation.

> JavaScript will call `toString()` or `toLocaleString()` on an object when it is converted to a string, such as when `+` adding to a string or in `${}` template literals.

Reviewer: I think JavaScript never implicitly calls `toLocaleString()`, right?

```suggestion
JavaScript will call `toString()` on an object when it is converted to a string, such as when `+` adding to a string or in `${}` template literals.
```

Author: You're absolutely right🙇‍♂️ I fixed it in 3ffe462.

The default Object `.toString()` or `toLocaleString()` uses the format `"[object Object]"`, which is often not what was intended.
This rule reports on stringified values that aren't primitives and don't define a more useful `.toString()` or `toLocaleString()` method.

> > Note that `Function` provides its own `.toString()` or `toLocaleString()` that returns the function's code.

Reviewer: ```suggestion
> Note that `Function` provides its own `.toString()` and `toLocaleString()` that return the function's code.
```

> See **https://typescript-eslint.io/rules/no-base-to-string** for documentation.

JavaScript will call `toString()` on an object when it is converted to a string, such as when `+` adding to a string or in `${}` template literals.

> The default Object `.toString()` or `toLocaleString()` uses the format `"[object Object]"`, which is often not what was intended.

Reviewer 2: ```suggestion
The default Object `.toString()` and `toLocaleString()` use the format `"[object Object]"`, which is often not what was intended.
```

### `ignoredTypeNames`

Stringified regular expressions of type names to ignore.
This is useful for types missing `toString()` or `toLocaleString()` (but actually has `toString()` or `toLocaleString()`).

> There are some types missing `toString()` or `toLocaleString()` in old version TypeScript, like `RegExp`, `URL`, `URLSearchParams` etc.

Reviewer 2: This is an existing typo, but since you're in the area, do you mind? 😄 

```suggestion
There are some types missing `toString()` or `toLocaleString()` in old versions of TypeScript, like `RegExp`, `URL`, `URLSearchParams` etc.
```

Author: Of course! I fixed it in ebc6947.
